shared/actions/teams/index.js

```diff
@@ -133,42 +133,66 @@ const _editMembership = function*(action: Constants.EditMembership) {
-  const {payload: {name, username, email}} = action
+  const {payload: {name, username, email, id}} = action
 
   yield put(
-    replaceEntity(['teams', 'teamNameToLoadingInvites'], I.Map([[name, I.Map([[username || email, true]])]]))
+    replaceEntity(
+      ['teams', 'teamNameToLoadingInvites'],
+      I.Map([[name, I.Map([[username || email || id, true]])]])
+    )
   )
 
-  // disallow call with both username & email
-  if (!!username && !!email) {
-    const errMsg = 'Supplied both email and username to removeMemberOrPendingInvite'
+  // disallow call with any pair of username, email, and ID to avoid black-bar errors
+  if ((!!username && !!email) || (!!username && !!id) || (!!email && !!id)) {
+    const errMsg = 'Supplied more than one form of identification to removeMemberOrPendingInvite'
     console.error(errMsg)
     throw new Error(errMsg)
   }
 
   yield put(replaceEntity(['teams', 'teamNameToLoading'], I.Map([[name, true]])))
   try {
-    yield call(RpcTypes.teamsTeamRemoveMemberRpcPromise, {param: {email, name, username}})
+    yield call(RpcTypes.teamsTeamRemoveMemberRpcPromise, {param: {email, name, username, inviteID: id}})
   } finally {
     yield put((dispatch: Dispatch) => dispatch(Creators.getDetails(name))) // getDetails will unset loading
     yield put(
       replaceEntity(
         ['teams', 'teamNameToLoadingInvites'],
-        I.Map([[name, I.Map([[username || email, false]])]])
+        I.Map([[name, I.Map([[username || email || id, false]])]])
       )
     )
   }
 }
 
 const _inviteToTeamByPhone = function*(action: Constants.InviteToTeamByPhone) {
-  const {payload: {teamname, phoneNumber}} = action
+  const {payload: {teamname, role, phoneNumber, fullName = ''}} = action
   yield put(
     replaceEntity(['teams', 'teamNameToLoadingInvites'], I.Map([[teamname, I.Map([[phoneNumber, true]])]]))
   )
-  openSMS(phoneNumber, 'delicious seitan') // TODO replace with token from seitan call
-  yield put(
-    replaceEntity(['teams', 'teamNameToLoadingInvites'], I.Map([[teamname, I.Map([[phoneNumber, false]])]]))
-  )
+  const seitan = yield call(RpcTypes.teamsTeamCreateSeitanTokenRpcPromise, {
+    param: {
+      name: teamname,
+      role: role && RpcTypes.TeamsTeamRole[role],
+      label: {t: 1, sms: {f: fullName, n: phoneNumber}},
+    },
+  })
+
+  /* Open SMS */
+  // seitan is 16chars
+  // message sans teamname is 134chars long. Absolute max teamname can be is 26chars to fit within 160 sms limit
+  // limit teamname to 15chars (+3 for ellipsis) to leave some wiggle room
+  let clippedTeamname = teamname
+  if (clippedTeamname.length > 18) {
+    clippedTeamname = `...${teamname.substring(teamname.length - 15)}`
+  }
+  const bodyText = `Please join the ${clippedTeamname} team on Keybase. Install and paste this in the "Teams" tab:\n\ntoken: ${seitan.toUpperCase()}\n\nquick install: keybase.io/_/go`
+  openSMS(phoneNumber, bodyText)
```

Comment:
this has pretty inconsistent behavior on iOS when the messages app isn't running in the background. likely a quirk of iOS's half-support of opening SMS links. can likely be made better by using MessageUI in the future.